shell/packages/sandstorm-ui-autocomplete-input/autocomplete-client.js

```diff
@@ -0,0 +1,210 @@
+        name: currentText,
+        service: "email",
+        intrinsicName: "Email address"
+      },
+      isDefault: true,
+    });
+  }
+  currentText = currentText.toLowerCase();
+  var selectedContactsIds = template.selectedContactsIds.get();
+  var contacts = UserContacts.find({_id: {$nin: selectedContactsIds}}).fetch();
+  var results;
+  if (currentText.lastIndexOf("@", 0) === 0) {
+    var textWithoutAt = currentText.slice(1);
+    results = _.filter(contacts, function (contact) {
+      return contact.profile.handle.toLowerCase().indexOf(textWithoutAt) !== -1;
+    });
+  } else {
+    results = _.filter(contacts, function (contact) {
+      return contact.profile.name.toLowerCase().indexOf(currentText) !== -1 ||
+        contact.profile.handle.toLowerCase().indexOf(currentText) !== -1 ||
+        contact.profile.intrinsicName.toLowerCase().indexOf(currentText) !== -1;
+    });
+  }
+  results.forEach(function (contact) {
+    var oldId = contact._id;
+    SandstormDb.fillInPictureUrl(contact);
+  })
+  template.autoCompleteContacts.set(defaults.concat(results));
+};
+
+Template.contactInputBox.helpers({
+  completedContacts: function () {
+    return Template.instance().selectedContacts.get();
+  },
+  autoCompleteContacts: function () {
+    return Template.instance().autoCompleteContacts.get();
+  },
+  inputActive: function () {
+    return Template.instance().inputActive.get();
+  },
+  isCurrentlySelected: function () {
+    var selectedContactId = Template.instance().highlightedContact.get()._id;
+
+    return selectedContactId === this._id;
+  },
+  templateId: function () {
+    return Template.instance().randomId;
+  }
+});
+
+function selectContact(template, highlightedContact, inputBox) {
+  if (highlightedContact.isDefault) {
+    if (highlightedContact.profile.service === "email") {
+      highlightedContact._id = inputBox.value;
+      highlightedContact.profile.name = inputBox.value;
+      highlightedContact.profile.pictureUrl = "/email.svg";
+    }
+  }
+  var contacts = template.selectedContacts.get();
+  contacts.push(highlightedContact);
+  template.selectedContacts.set(contacts);
+
+  var selectedContactsIds = template.selectedContactsIds.get();
+  selectedContactsIds.push(highlightedContact._id);
+  template.selectedContactsIds.set(selectedContactsIds);
+
+  template.highlightedContact.set({_id: null});
+  inputBox.value = "";
+  template.currentText.set(null);
+}
+function deleteSelected(contact, template) {
+  var self = contact;
+  var contacts = template.selectedContacts.get();
+  template.selectedContacts.set(_.filter(contacts, function (contact) {
+    return contact._id !== self._id;
+  }));
+
+  var selectedContactsIds = template.selectedContactsIds.get();
+  template.selectedContactsIds.set(_.filter(selectedContactsIds, function (id) {
+    return id !== self._id;
+  }));
+  template.find("input").focus();
+}
+
+Template.contactInputBox.events({
+  // "click .contact-box": function (event) {
```

Comment:
What is the status of this commented-out code? At the very least, it should have a comment about why it is here.